Lombiq.HelpfulLibraries/Libraries/Navigation/PortalNavigationProviderBase.cs

```diff
@@ -0,0 +1,45 @@
+﻿using Lombiq.HelpfulLibraries.Libraries.Utilities;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.Localization;
+using OrchardCore.Navigation;
+using System;
+using System.Threading.Tasks;
+
+namespace Lombiq.HelpfulLibraries.Libraries.Navigation
+{
+    public abstract class PortalNavigationProviderBase : INavigationProvider
+    {
+        public const string NavigationName = "portal";

```

Comment:
In addition to my comment regarding widget name, I meant this navigation name to be SNOW-specific. If it's a more generic one, I'll leave it to you to decide, could be something like `site`. Or you can make it adjustable from the inherited class.

Reply:
I thought with "portal" we are creating a standard name for main menu front end navigation providers. I don't see a reason to get cold feet now, portal is as good a name as site.